machine_translation/prepare_data.py

```diff
@@ -160,7 +161,7 @@ def merge_parallel(src_filename, trg_filename, merged_filename):
                 while True:
                     lline = left.readline()
                     rline = right.readline()
-                    if (lline == '') or (rline == ''):
+                    if (lline == '') and (rline == ''):
```

Comment:
actually the next condition handles that one. I've tried shuffled files that reportedly causing crashes, but couldn't replicate them, and eyeballed the sections that cause the error, there were no empty lines etc. I tracked it down to the point that, it is crashing at the end of an epoch, probably cannot reset the stream, returning empty batch. Also another common thing among people experiencing these crashes, is that, they've installed `blocks` with pip, i've checked one of such codebases and `fuel` was different from mine at least. So bottom-line, i think that error is caused by some change/mechanism in `fuel`.